jocd/src/br/org/certi/jocd/dapaccess/Transfer.java

```diff
@@ -47,11 +64,12 @@ public int getDataSize() {
   public void addResponse(byte[] data) {
     assert data.length == this.sizeBytes;
     int resultSize = this.sizeBytes / 4;
-    int[] result = new int[resultSize];
+    byte[] result = new byte[resultSize];
     int count = 0;
     for (int i = 0; i < this.sizeBytes; i += 4) {
-      int word = ((data[0 + i] << 0) | (data[1 + i] << 8) | (data[2 + i] << 16) | (data[3 + i]
-          << 24));
+      byte word = (byte) (((data[0 + i] << 0) | (data[1 + i] << 8) | (data[2 + i] << 16) | (
```

Comment:
The result of this logical operation is a 4-byte value, it should not be stored in a byte variable.